src/h5web/visualizations/line/hooks.ts

```diff
@@ -4,30 +4,16 @@ import { useThree } from 'react-three-fiber';
 import { Domain } from '../shared/models';
 import { findDomain, getAxisScale } from '../shared/utils';
 import { AxisSystemContext } from '../shared/AxisSystemProvider';
-import {
-  useVisProps,
-  useFlatValues,
-} from '../../dataset-visualizer/VisProvider';
+import { useVisProps } from '../../dataset-visualizer/VisProvider';
 
 export function useData(): number[] {
-  const { rawValues, rawDims, mapping } = useVisProps();
-  const values = useFlatValues();
+  const { visValues, slicingIndices } = useVisProps();
 
-  if (rawDims.length === 1) {
-    return rawValues;
+  if (slicingIndices) {
+    return (visValues as number[][])[slicingIndices[0]];
   }
```

Comment:
If we need to slice, it is always in the shallowest dimension as we arrange `visValues` as such. If not, `slicingIndices` are undefined and we return the raw values.